### PR TITLE
Zip file Retry

### PR DESCRIPTION
**Pull request by saquino0827**
## Description

- Failure to unzip files now no longer deletes the file from the SFTP Server
  - With this change, our cron will automagically retry unzipping. This will mean that a fix will need to be made in between the cron schedule to make unzipping work again.

## Issue

[#1558](https://github.com/CDCgov/trusted-intermediary/issues/1558)

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)

### CODE REVIEW

File: src/sftp/handler.go [hunk 1]
```diff
@@ -238,7 +238,9 @@ func (receiver *SftpHandler) copySingleFile(fileInfo os.FileInfo, index int, dir
 
 	slog.Info("About to consider whether this is a zip", slog.String(utils.FileNameKey, fileInfo.Name()))
 
-	if strings.Contains(fileInfo.Name(), ".zip") {
+	deleteZip := false
```
Comment (pluckyswan) [inline src/sftp/handler.go:241]: deleteZipFolder instead /[-_-}\

**Review by jcrichlake — APPROVED** [review]

File: src/sftp/handler.go (hunk 1 above)
Comment (saquino0827) [reply src/sftp/handler.go]: Sounds good, change has been made

File: src/sftp/handler.go [hunk 2]
```diff
@@ -258,11 +262,12 @@ func (receiver *SftpHandler) copySingleFile(fileInfo os.FileInfo, index int, dir
 		}
 	}
 
-	err = receiver.sftpClient.Remove(fullFilePath)
-	if err != nil {
-		slog.Error("Failed to remove file from SFTP server", slog.Any(utils.ErrorKey, err), slog.String(utils.FileNameKey, fullFilePath))
-		return
+	if !isZip || (isZip && deleteZip) {
```
[the remark this answers is not in the document]
Comment (saquino0827) [reply src/sftp/handler.go]: Yeah, this could be simplified more and has been implemented.

**Review by pluckyswan — APPROVED** [review]
Me likey

File: src/sftp/handler.go [hunk 3]
```diff
@@ -258,11 +262,12 @@ func (receiver *SftpHandler) copySingleFile(fileInfo os.FileInfo, index int, dir
 		}
 	}
 
-	err = receiver.sftpClient.Remove(fullFilePath)
-	if err != nil {
-		slog.Error("Failed to remove file from SFTP server", slog.Any(utils.ErrorKey, err), slog.String(utils.FileNameKey, fullFilePath))
-		return
+	if !isZip || deleteZip {
```
Comment (somesylvie) [inline src/sftp/handler.go:265]: It would be nice to have a test around this new condition

Comment (saquino0827) [reply src/sftp/handler.go:265]: I ran code with coverage and it say's that it is. I believe it's covered but just making sure before I merge/add more tests.
Screenshot 2024-11-07 at 12 01 55 PM

Comment (saquino0827) [reply src/sftp/handler.go:265]: I do see other scenarios that aren't covered that aren't part of the `if !isZip || deleteZip`. If that's what you meant, then I can look into adding coverage for those.

Comment (saquino0827) [reply src/sftp/handler.go:265]: Added an assert for making sure **Remove** is not called when a zip file is not successfully unzipped.

**Review by somesylvie — APPROVED** [review]

**Pull request merged**